src/components/Chat.tsx

```diff
@@ -316,9 +315,10 @@ const Chat: React.FC<ChatProps> = ({
   };
 
   useEffect(() => {
-    async function handleSelection(selectedText: string) {
-      const wordCount = selectedText.split(' ').length;
-      const tokenCount = await chainManager.chatModelManager.countTokens(selectedText);
+    async function handleSelection(event: CustomEvent) {
+      console.log('handleSelection1', event);
```

Comment:
Don't forget to remove console.logs once you are confident it works, same for the ones below.